--- InsightToolkit-3.20.0/Utilities/MetaIO/metaUtils.cxx.orig	2012-07-08 22:18:56.849178039 +0400
+++ InsightToolkit-3.20.0/Utilities/MetaIO/metaUtils.cxx	2012-07-08 22:22:12.243169830 +0400
@@ -797,16 +797,16 @@
 //
 bool MET_StringToWordArray(const char *s, int *n, char ***val)
 {
-  ptrdiff_t l = strlen(s);
+  std::ptrdiff_t l = strlen(s);
 
-  ptrdiff_t p = 0;
+  std::ptrdiff_t p = 0;
   while(p<l && s[p] == ' ')
     {
     p++;
     }
 
   *n = 0;
-  ptrdiff_t pp = p;
+  std::ptrdiff_t pp = p;
   bool space = false;
   while(pp<l)
     {
@@ -837,7 +837,7 @@
 
   *val = new char *[*n];
 
-  ptrdiff_t i, j;
+  std::ptrdiff_t i, j;
   for(i=0; i<*n; i++)
     {
     if(p == l)
--- InsightToolkit-3.20.0/Code/IO/itkImageIORegion.h.orig	2012-07-08 22:48:32.299211829 +0400
+++ InsightToolkit-3.20.0/Code/IO/itkImageIORegion.h	2012-07-08 22:51:10.637205170 +0400
@@ -59,8 +59,8 @@
 
   /** these types correspond to those of itk::Size, itk::Offset and itk::Index */
   typedef size_t                  SizeValueType;
-  typedef ptrdiff_t               IndexValueType;
-  typedef ptrdiff_t               OffsetValueType;
+  typedef std::ptrdiff_t               IndexValueType;
+  typedef std::ptrdiff_t               OffsetValueType;
 
 
   /** Index typedef support. An index is used to access pixel values. */
--- InsightToolkit-3.20.0/Code/IO/itkConvertPixelBuffer.txx.orig	2012-07-08 23:05:34.094168852 +0400
+++ InsightToolkit-3.20.0/Code/IO/itkConvertPixelBuffer.txx	2012-07-08 23:06:04.297167520 +0400
@@ -252,7 +252,7 @@
     // http://www.inforamp.net/~poynton/notes/colour_and_gamma/ColorFAQ.html
     // NOTE: The scale factors are converted to whole numbers for
     // precision
-    ptrdiff_t diff = inputNumberOfComponents - 4;
+    std::ptrdiff_t diff = inputNumberOfComponents - 4;
     InputPixelType* endInput = inputData + size * (size_t)inputNumberOfComponents;
     while(inputData != endInput)
       {
--- InsightToolkit-3.20.0/Code/IO/itkConvertPixelBuffer.txx.orig	2012-07-08 23:19:26.816133839 +0400
+++ InsightToolkit-3.20.0/Code/IO/itkConvertPixelBuffer.txx	2012-07-08 23:20:19.956131692 +0400
@@ -381,7 +381,7 @@
   // just skip the rest of the data
   else
     {
-    ptrdiff_t diff = inputNumberOfComponents - 3;
+    std::ptrdiff_t diff = inputNumberOfComponents - 3;
     InputPixelType* endInput = inputData + size * (size_t)inputNumberOfComponents;
     while(inputData != endInput)
       {
@@ -519,7 +519,7 @@
     }
   else
     {
-    ptrdiff_t diff = inputNumberOfComponents - 4;
+    std::ptrdiff_t diff = inputNumberOfComponents - 4;
     InputPixelType* endInput = inputData + size * (size_t)inputNumberOfComponents;
     while(inputData != endInput)
       {
@@ -661,7 +661,7 @@
                               OutputPixelType* outputData , 
                               size_t size)
 {
-  ptrdiff_t diff = inputNumberOfComponents - 2;
+  std::ptrdiff_t diff = inputNumberOfComponents - 2;
   InputPixelType* endInput = inputData + size * (size_t)inputNumberOfComponents;
   while(inputData != endInput)
     {
--- InsightToolkit-3.20.0/Testing/Code/Common/itkNumericTraitsTest.cxx.orig	2012-07-08 23:37:03.940089398 +0400
+++ InsightToolkit-3.20.0/Testing/Code/Common/itkNumericTraitsTest.cxx	2012-07-08 23:42:34.271075485 +0400
@@ -19,6 +19,7 @@
 #endif
 #include <iostream>
 #include <vector>
+#include <cstddef>
 
 #include "itkNumericTraits.h"
 
